Modules/_asynciomodule.c

```diff
@@ -2361,9 +2317,64 @@ swap_current_task(asyncio_state *state, PyObject *loop, PyObject *task)
     Py_BEGIN_CRITICAL_SECTION(current_tasks);
     prev_task = swap_current_task_lock_held(current_tasks, loop, hash, task);
     Py_END_CRITICAL_SECTION();
+    set_ts_asyncio_running_task(loop, task);
     return prev_task;
 }
 
+static inline void
+set_ts_asyncio_running_task(PyObject *loop, PyObject *task)
+{
+    // This is needed to enable `asyncio.capture_call_graph()` API.
+    // We want to be enable debuggers and profilers to be able to quickly
+    // introspect the asyncio running state from another process.
+    // When we do that, we need to essentially traverse the address space
+    // of a Python process and understand what every Python thread in it is
+    // currently doing, mainly:
+    //
+    //  * current frame
+    //  * current asyncio task
+    //
+    // A naive solution would be to require profilers and debuggers to
+    // find the current task in the "_asynciomodule" module state, but
+    // unfortunately that would require a lot of complicated remote
+    // memory reads and logic, as Python's dict is a notoriously complex
+    // and ever-changing data structure.
+    //
+    // So the easier solution is to put a strong reference to the currently
+    // running `asyncio.Task` on the interpreter thread state (we already
+    // have some asyncio state there.)
```

Comment:
```suggestion
    // We want to be enable debuggers and profilers to be able to quickly
    // introspect the asyncio running state from another process.
    // When we do that, we need to essentially traverse the address space
    // of a Python process and understand what every Python thread in it is
    // currently doing, mainly:
    //
    //  * current frame
    //  * current asyncio task
    //
    // A naive solution would be to require profilers and debuggers to
    // find the current task in the "_asynciomodule" module state, but
    // unfortunately that would require a lot of complicated remote
    // memory reads and logic, as Python's dict is a notoriously complex
    // and ever-changing data structure.
    //
    // So the easier solution is to put a strong reference to the currently
    // running `asyncio.Task` on the current thread state (the current loop
    // is also stored there.)
```
this is only needed to external introspection IIUC as such the comment of `asyncio.capture_call_graph` isn't needed.